controllers/yawol-controller/loadbalancerset/helper.go

```diff
@@ -0,0 +1,94 @@
+package loadbalancerset
+
+import (
+	"fmt"
+
+	yawolv1beta1 "github.com/stackitcloud/yawol/api/v1beta1"
+	"github.com/stackitcloud/yawol/internal/helper"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+var relevantLBMConditionslForLBS = sets.New[helper.LoadbalancerCondition](
+	helper.ConfigReady,
+	helper.EnvoyReady,
+	helper.EnvoyUpToDate,
+)
+
+// areRelevantConditionsMet returns if all required conditions (from the
+// perspective of the LoadBalancerSet) are both `True` and up-to-date, according
+// to the passed expiration time.
+func areRelevantConditionsMet(machine *yawolv1beta1.LoadBalancerMachine, expiration metav1.Time, checkTransition bool) (bool, error) {
+	if machine.Status.Conditions == nil {
+		return false, fmt.Errorf("no conditions set")
+	}
+	found := 0
+	for _, condition := range *machine.Status.Conditions {
+		if !relevantLBMConditionslForLBS.Has(helper.LoadbalancerCondition(condition.Type)) {
+			continue
+		}
+		found++
+		transitionCheck := true
+		if checkTransition {
+			transitionCheck = condition.LastTransitionTime.Before(&expiration)
```

Comment:
Why do we need to check the transition time in `shouldMachineBeDeleted`?

Reply:
this is replicating the previous logic, where condition was only considered "failed" if it has been `false` for some time.